### PR TITLE
add eject_model, get_loaded_models, get_cuda_info, get_cpu_info apis

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
features

* **What is the current behavior?** (You can also link to an open issue here)
no model cache management features

* **What is the new behavior (if this is a feature change)?**
we can now manage the model cache in marqo

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:
this PR is related to a PR in marqo. [link](https://github.com/marqo-ai/marqo/pull/239)

### CODE REVIEW

**Review by pandu-k — CHANGES_REQUESTED** [review]
Please add a few simple tests for each endpoint

**Comment by wanliAlex** [comment]
> Please add a few simple tests for each endpoint

added

File: tests/v0_tests/test_model_cache_management.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from unittest import mock
+
+
+class TestModelCacheManagement(MarqoTestCase):
+
+    # NOTE: The cuda should already have model loaded in the startup
+    def setUp(self) -> None:
+        self.client = Client(**self.client_settings)
+        self.index_name = "test_index"
+        self.MODEL = "ViT-L/14"
```
Comment (pandu-k) [inline tests/v0_tests/test_model_cache_management.py]: Can we set this to the smaller CLIP model? Sometimes test instances can't handle the memory footprint of this large model

File: tests/v0_tests/test_model_cache_management.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.client.get_cpu_info()
+
+
+    def test_get_loaded_models(self) -> None:
+        self.client.get_loaded_models()
+
+
+    def test_eject_no_cached_model(self) -> None:
+        try:
+            self.client.eject_model("void_model", "void_device")
```
Comment (pandu-k) [inline tests/v0_tests/test_model_cache_management.py]: What does this test do?

**Review by pandu-k — CHANGES_REQUESTED** [review]
Can you make the unit tests slightly more detailed?

For example, testing for the structure of the response.

File: tests/v0_tests/test_model_cache_management.py (hunk 1 above)
Comment (wanliAlex) [reply tests/v0_tests/test_model_cache_management.py]: we change to "ViT-B/32" .

File: tests/v0_tests/test_model_cache_management.py (hunk 2 above)
Comment (wanliAlex) [reply tests/v0_tests/test_model_cache_management.py]: try to eject a model that is not cached. comments added.

**Comment by wanliAlex** [comment]
> Can you make the unit tests slightly more detailed?
> 
> For example, testing for the structure of the response.

we can't verify the correctness of the response. so we just check if the keys are in the responded dictionary here.

**Comment by wanliAlex** [comment]
the tests are added and passed. 
image

File: tests/v0_tests/test_model_cache_management.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        res = self.client.get_loaded_models()
+
+        if "models" not in res:
+            raise AssertionError
+
+
+    def test_eject_no_cached_model(self) -> None:
+        # test a model that is not cached
+        try:
+            res = self.client.eject_model("void_model", "void_device")
```
Comment (pandu-k) [inline tests/v0_tests/test_model_cache_management.py:66]: Raise an assertion error below this line: if we successfully eject a non-existent model without an error being thrown, that is a problem!

**Review by pandu-k — CHANGES_REQUESTED** [review]

**Comment by wanliAlex** [comment]
the latest version can pass the test.

image

**Review by pandu-k — APPROVED** [review]

**Pull request merged**